IdentityCore/tests/MSIDAccountMetadataCacheItemTests.m

```diff
@@ -47,10 +47,24 @@ - (void)testJSONDictionary_whenAllFieldsSet_shouldReturnJSONDictionaryWithAccoun
     NSError *error;
     
     XCTAssertTrue([cacheItem setCachedURL:[NSURL URLWithString:@"https://contoso.com"]
-                            forRequestURL:[NSURL URLWithString:@"https://testAuthority.com"] error:&error]);
+                            forRequestURL:[NSURL URLWithString:@"https://testAuthority.com"]
+                            instanceAware:NO
+                                    error:&error]);
+    XCTAssertNil(error);
+
+    XCTAssertTrue([cacheItem setCachedURL:[NSURL URLWithString:@"https://contoso3.com"]
+                            forRequestURL:[NSURL URLWithString:@"https://testAuthority3.com"]
+                            instanceAware:YES
+                                    error:&error]);
+    XCTAssertNil(error);
+
+    XCTAssertTrue([cacheItem setCachedURL:[NSURL URLWithString:@"https://contoso2.com"]
+                            forRequestURL:[NSURL URLWithString:@"https://testAuthority2.com"]
+                            instanceAware:NO
+                                    error:&error]);
     XCTAssertNil(error);
     
-    __auto_type *expected = @{ @"account_metadata" : @{ @"URLMap" : @{ @"https://testAuthority.com" : @"https://contoso.com"} },
+    __auto_type *expected = @{ @"account_metadata" : @{ @"URLMap" : @{@"instance_aware-NO" : @{ @"https://testAuthority.com" : @"https://contoso.com", @"https://testAuthority2.com" : @"https://contoso2.com"}, @"instance_aware-YES" :  @{ @"https://testAuthority3.com" : @"https://contoso3.com"}} },
```

Comment:
What if in future we need to store additional parameters along with "instance_aware"? I assume we will modify the key, so it would be something like: `"instance_aware-NO&qwe-asd",` is it correct?

Reply:
I changed the schema a little bit. Yes, the idea is similar :)